body/test/body.test.ts

```diff
@@ -0,0 +1,342 @@
+        try {
+          const body = await readBody(res, { limit: 1024 });
+          expect(body).toEqual("missing `removeListener` function");
+        } finally {
+          server.close(done);
+        }
+      });
+    });
+  });
+
+  test("should error if the stream is not readable", (done) => {
+    const server = http.createServer(async (req, res) => {
+      try {
+        const reqNoOn = {
+          on: req.on,
+          removeListener: req.removeListener,
+          readable: false,
+        };
+        const body = await readBody(reqNoOn, { limit: 1024 });
+        console.dir(body);
+        res.end(body);
+      } catch (err) {
+        req.resume();
+        res.statusCode = 500;
+        if (err instanceof Error) {
+          return res.end(err.message);
+        } else {
+          return res.end("unknown error");
+        }
+      }
+    });
+
+    server.listen(() => {
+      const addr = server.address() as AddressInfo;
+      const client = http.request({ method: "POST", port: addr.port });
+
+      client.end("hello, world!");
+
+      client.on("response", async (res) => {
+        try {
+          const body = await readBody(res, { limit: 1024 });
+          expect(body).toEqual("stream is not readable");
+        } finally {
+          server.close(done);
+        }
+      });
+    });
+  });
+
+  test("should error if limit is not present", (done) => {
+    const server = http.createServer(async (req, res) => {
+      try {
+        const reqNoOn = {
+          on: req.on,
+          removeListener: req.removeListener,
+          readable: req.readable,
+        };
+        const body = await readBody(reqNoOn, { limit: undefined as any });
+        console.dir(body);
+        res.end(body);
+      } catch (err) {
+        req.resume();
+        res.statusCode = 500;
+        if (err instanceof Error) {
+          return res.end(err.message);
+        } else {
+          return res.end("unknown error");
+        }
+      }
+    });
+
+    server.listen(() => {
+      const addr = server.address() as AddressInfo;
+      const client = http.request({ method: "POST", port: addr.port });
+
+      client.end("hello, world!");
+
+      client.on("response", async (res) => {
+        try {
+          const body = await readBody(res, { limit: 1024 });
+          expect(body).toEqual("must set a limit");
+        } finally {
+          server.close(done);
+        }
+      });
+    });
+  });
+
+  test("should timeout if no chunks are sent", async () => {
+    const stream = {
+      on: () => undefined,
+      removeListener: () => undefined,
+      readable: true,
+    };
+
+    try {
+      await readBody(stream, { limit: 100 });
+      throw new Error("this should not return successfully");
+    } catch (e) {
+      expect(e).toEqual(new Error("received no body chunks after 100ms"));
+    }
+
+    // setting test timeout to 105 because the promise should throw after 100ms
+  }, 105);
```

Comment:
```suggestion
  }, 150);
```
Timers in JS can drift. I'd set to at least 150 but maybe even 200 or 250